src/ReturnTypes/ModelExtension.php

```diff
@@ -109,6 +110,10 @@ public function getTypeFromStaticMethodCall(
             $returnType = new GenericObjectType(EloquentBuilder::class, [new ObjectType($scope->resolveName($methodCall->class))]);
         }
 
+        if (in_array(Collection::class, $returnType->getReferencedClasses()) && $methodCall->name->name === 'all') {
```

Comment:
We should use `$methodReflection->getName()` instead of `$methodCall->name->name`

Also, can you check if we need to do this to methods other than `all` ?

Reply:
I think `all` is the only one that should be fixed in the `ModelExtension.php`. However when adding the test below it gives the same error. So maybe I'm working in the wrong file?

```php
public static function findFirstFooBarThread(): Thread
{
    return self::query()->where('foo', 'bar')->get()->first();
}
```

Reply:
That `get` call is on the builder not model. So let's make another issue/PR for that one.

Reply:
Will do!